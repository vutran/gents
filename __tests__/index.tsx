import gents from '../src';

describe('gents', () => {
    describe('basic types', () => {
        it('should convert a number', () => {
            const source = '{ "user_id": 1 }';
            const def = gents(source, 'User');
            expect(def).toMatchSnapshot();
        });
        it('should convert a string', () => {
            const source = '{ "name": "Vu Tran" }';
            const def = gents(source, 'User');
            expect(def).toMatchSnapshot();
        });
        it('should convert a Array of numbers', () => {
            const source = '{ "user_ids": [10, 20, 30] }';
            const def = gents(source, 'FooGroup');
            expect(def).toMatchSnapshot();
        });
        it('should convert a Array of strings', () => {
            const source =
                '{ "locations": ["San Francisco", "Los Angeles", "New York"] }';
            const def = gents(source, 'FooGroup');
            expect(def).toMatchSnapshot();
        });
        it('should convert a mixed Array of numbers and strings', () => {
            const source = '{ "mixed_items": ["apple", 1, "banana", 2] }';
            const def = gents(source, 'MixedGroup');
            expect(def).toMatchSnapshot();
        });
    });

    describe('subtypes', () => {
        it('should create subtypes for Arrays', () => {
            const source =
                '{ "locations": ["San Francisco", "Los Angeles", "New York"] }';
            const def = gents(source, 'FooGroup', { subtypes: true });
            expect(def).toMatchSnapshot();
        });
        it('should create multiple subtypes', () => {
            const source =
                '{ "locations": ["San Francisco", "Los Angeles", "New York"], "user_ids": [10, 20, 30] }';
            const def = gents(source, 'FooGroup', { subtypes: true });
            expect(def).toMatchSnapshot();
        });
    });
});
