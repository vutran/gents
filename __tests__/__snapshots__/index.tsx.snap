// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gents basic types should convert a Array of numbers 1`] = `
"interface FooGroup {
    user_ids: (number)[];
}"
`;

exports[`gents basic types should convert a Array of strings 1`] = `
"interface FooGroup {
    locations: (string)[];
}"
`;

exports[`gents basic types should convert a mixed Array of numbers and strings 1`] = `
"interface MixedGroup {
    mixed_items: (string | number)[];
}"
`;

exports[`gents basic types should convert a number 1`] = `
"interface User {
    user_id: number;
}"
`;

exports[`gents basic types should convert a string 1`] = `
"interface User {
    name: string;
}"
`;

exports[`gents subtypes should create multiple subtypes 1`] = `
"interface FooGroup {
    locations: Locations[];
    user_ids: UserIds[];
}
type Locations = string;
type UserIds = number;"
`;

exports[`gents subtypes should create subtypes for Arrays 1`] = `
"interface FooGroup {
    locations: Locations[];
}
type Locations = string;"
`;
